server/services/interventionsService.test.ts

```diff
@@ -2499,6 +2500,233 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
+      const fetchedAppointment = await interventionsService.getSupplierAssessmentAppointment(
+        token,
+        '80fb9e6d-1d0e-4204-9c5f-f86970d25c50'
+      )
+      expect(fetchedAppointment.appointmentTime).toEqual('2021-05-13T12:30:00Z')
+      expect(fetchedAppointment.durationInMinutes).toEqual(120)
+    })
+  })
+
+  describe('updateSupplierAssessmentAppointment', () => {
+    describe('with non-null values', () => {
+      it('returns an updated supplier assessment appointment', async () => {
+        const appointment = appointmentFactory.build({
+          appointmentTime: '2021-05-13T12:30:00Z',
+          durationInMinutes: 60,
+        })
+
+        await provider.addInteraction({
+          state: 'a sent referral with ID fae5df9d-41a9-4be6-98f7-0f1378e85ba1 exists',
+          uponReceiving:
+            'a PATCH request to update the change the duration to an hour for the initial assessment appointment of the sent referral with ID fae5df9d-41a9-4be6-98f7-0f1378e85ba1',
+          withRequest: {
+            method: 'PATCH',
+            path: '/sent-referral/fae5df9d-41a9-4be6-98f7-0f1378e85ba1/supplier-assessment-appointment',
+            body: {
+              durationInMinutes: 60,
+            },
+            headers: { Accept: 'application/json', Authorization: `Bearer ${token}` },
+          },
+          // note - this is an exact match
+          willRespondWith: {
+            status: 200,
+            body: appointment,
+            headers: {
+              'Content-Type': 'application/json',
+            },
+          },
+        })
+
+        expect(
+          await interventionsService.updateSupplierAssessmentAppointment(
+            token,
+            'fae5df9d-41a9-4be6-98f7-0f1378e85ba1',
+            {
+              durationInMinutes: 60,
+            }
+          )
+        ).toMatchObject(appointment)
+      })
+    })
+  })
+
+  describe('recordSupplierAssessmentAppointmentAttendance', () => {
+    it('returns an updated supplier assessment appointment with the service user‘s attendance', async () => {
+      await provider.addInteraction({
+        state:
+          'a sent referral with ID 060946a4-1574-4581-b539-5df69303c595 exists and no feedback has been recorded for its supplier assessment appointment',
+        uponReceiving:
+          'a POST request to set the attendance for the supplier assessment appointment on sent referral with ID 060946a4-1574-4581-b539-5df69303c595',
+        withRequest: {
+          method: 'POST',
+          path: '/sent-referral/060946a4-1574-4581-b539-5df69303c595/supplier-assessment-appointment/record-attendance',
+          body: {
+            attended: 'late',
+            additionalAttendanceInformation: 'Alex missed the bus',
+          },
+          headers: { Accept: 'application/json', Authorization: `Bearer ${token}` },
+        },
+        willRespondWith: {
+          status: 200,
+          body: Matchers.like({
+            sessionNumber: 2,
```

Comment:
Supplier assessments do not include sessions

Reply:
D'oh, too eager on the copy and paste, thanks for catching that. I'll fix it.